### PR TITLE
add update, delete methods to book controller, add liquibase support, add dev controller for fast fill db purposes

### PR DESCRIPTION
**Pull request by ihorromaniuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -42,4 +45,16 @@ public ResponseEntity<?> createBook(@RequestBody CreateBookRequestDto requestDto
...
+    @PutMapping("/{id}")
+    public ResponseEntity<?> updateBook(@PathVariable Long id,
+                                        @RequestBody UpdateBookRequestDto requestDto) {
+        return ResponseEntity.ok(bookService.update(id, requestDto));
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> deleteBook(@PathVariable Long id) {
+        bookService.delete(id);
+        return ResponseEntity.noContent().build();
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java]:
```suggestion
    public void deleteBook(@PathVariable Long id) {
        bookService.delete(id);
```

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -42,4 +45,16 @@ public ResponseEntity<?> createBook(@RequestBody CreateBookRequestDto requestDto
...
                 .body(responseBook);
     }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<?> updateBook(@PathVariable Long id,
+                                        @RequestBody UpdateBookRequestDto requestDto) {
+        return ResponseEntity.ok(bookService.update(id, requestDto));
+    }
+
+    @DeleteMapping("/{id}")
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java:58]:
```suggestion
    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
```

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -42,4 +45,16 @@ public ResponseEntity<?> createBook(@RequestBody CreateBookRequestDto requestDto
                         .toString()))
                 .body(responseBook);
     }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<?> updateBook(@PathVariable Long id,
+                                        @RequestBody UpdateBookRequestDto requestDto) {
+        return ResponseEntity.ok(bookService.update(id, requestDto));
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java]:
```suggestion
    public BookDto updateBook(@PathVariable Long id,
                                        @RequestBody UpdateBookRequestDto requestDto) {
        return bookService.update(id, requestDto);
```

File: src/main/java/core/basesyntax/bookstore/controller/DevController.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/dev")
+@RequiredArgsConstructor
+public class DevController {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/controller/DevController.java]: remove this controller

File: src/main/java/core/basesyntax/bookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -35,5 +37,21 @@ public BookDto getById(Long id) {
...
     }
+
+    @Override
+    public void delete(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, UpdateBookRequestDto requestDto) {
+        if (bookRepository.existsById(id)) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(()....
        bookMapper.updateBookFromDto(requestDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

File: src/main/java/core/basesyntax/bookstore/mapper/BookMapper.java [hunk 6]
```diff
@@ -3,12 +3,15 @@
...
+import core.basesyntax.bookstore.dto.book.UpdateBookRequestDto;
 import core.basesyntax.bookstore.model.Book;
 import org.mapstruct.Mapper;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
-    Book toModel(CreateBookRequestDto dto);
+    Book createDtoToModel(CreateBookRequestDto dto);
+
+    Book updateDtoToModel(UpdateBookRequestDto dto);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/mapper/BookMapper.java]:
```suggestion
    void updateBookFromDto(UpdateBookRequestDto dto, @MappingTarget Book entity);
```

File: src/main/java/core/basesyntax/bookstore/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -35,5 +37,21 @@ public BookDto getById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id).orElseThrow(() ->
                 new EntityNotFoundException("Can't find book by id. Id: " + id)));
     }
+
+    @Override
+    public void delete(Long id) {
+        bookRepository.deleteById(id);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/service/impl/BookServiceImpl.java]: also, you can check if book exists

File: src/main/resources/db/changelog/changes/02-add-column-is-deleted.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+databaseChangeLog:
+  - changeSet:
+      id: add-column-is_deleted
+      author: Ihor Romaniuk
+      objectQuotingStrategy: QUOTE_ONLY_RESERVED_WORDS
+      changes:
+        - addColumn:
+            columns:
+              - column:
+                  name: is_deleted
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-add-column-is-deleted.yaml]: add this column to existing changelog 01 and remove this changeset

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/core/basesyntax/bookstore/controller/DevController.java (hunk 4 above)
Comment (ihorromaniuk) [reply src/main/java/core/basesyntax/bookstore/controller/DevController.java:16]: Why? I need it to speed up dev process

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**